src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerEndpoint.java

```diff
@@ -129,6 +125,26 @@ public KeyMaterialFactory newKeyMaterialFactory(@Nonnull Item context, @Nonnull
         return newKeyMaterialFactory(dotDocker, creds);
     }
 
+    /**
+     * Makes the key materials available locally and returns {@link KeyMaterialFactory} that gives you the parameters
+     * needed to access it.
+     */
+    public KeyMaterialFactory newKeyMaterialFactory(@Nonnull Run context, @Nonnull VirtualChannel target) throws IOException, InterruptedException {
+        DockerServerCredentials creds=null;
+        if (credentialsId!=null) {
+            List<DomainRequirement> domainRequirements = URIRequirementBuilder.fromUri(getUri()).build();
+            domainRequirements.add(new DockerServerDomainRequirement());
+            creds = CredentialsProvider.findCredentialById(credentialsId, DockerServerCredentials.class, context,
+                    domainRequirements);
+        }
+
+        // the directory needs to be outside workspace to avoid prying eyes
```

Comment:
If a new overload is indeed necessary, clearly we would want the duplicated code to be factored out into an internal method somehow.

Reply:
@jglick How do you want me to factor this out exactly ? I don't see how I could avoid to duplicate the code here unless I use a typing like `Object` or one of the interface that `Run` and `Item` have in common (not sure if doing this would make sense though). So is it worth it ?

Reply:
I guess it does not matter so long as only one nondeprecated code path remains.